src/NzbDrone.Core/Indexers/Definitions/XSpeeds.cs

```diff
@@ -0,0 +1,334 @@
+
+    public XSpeedsRequestGenerator(UserPassTorrentBaseSettings settings, IndexerCapabilities capabilities)
+    {
+        _settings = settings;
+        _capabilities = capabilities;
+    }
+
+    public IndexerPageableRequestChain GetSearchRequests(MovieSearchCriteria searchCriteria)
+    {
+        var pageableRequests = new IndexerPageableRequestChain();
+
+        pageableRequests.Add(GetPagedRequests($"{searchCriteria.SanitizedSearchTerm}", searchCriteria.Categories, searchCriteria.FullImdbId));
+
+        return pageableRequests;
+    }
+
+    public IndexerPageableRequestChain GetSearchRequests(MusicSearchCriteria searchCriteria)
+    {
+        var pageableRequests = new IndexerPageableRequestChain();
+
+        pageableRequests.Add(GetPagedRequests($"{searchCriteria.SanitizedSearchTerm}", searchCriteria.Categories));
+
+        return pageableRequests;
+    }
+
+    public IndexerPageableRequestChain GetSearchRequests(TvSearchCriteria searchCriteria)
+    {
+        var pageableRequests = new IndexerPageableRequestChain();
+
+        pageableRequests.Add(GetPagedRequests($"{searchCriteria.SanitizedTvSearchString}", searchCriteria.Categories, searchCriteria.FullImdbId));
+
+        return pageableRequests;
+    }
+
+    public IndexerPageableRequestChain GetSearchRequests(BookSearchCriteria searchCriteria)
+    {
+        var pageableRequests = new IndexerPageableRequestChain();
+
+        pageableRequests.Add(GetPagedRequests($"{searchCriteria.SanitizedSearchTerm}", searchCriteria.Categories));
+
+        return pageableRequests;
+    }
+
+    public IndexerPageableRequestChain GetSearchRequests(BasicSearchCriteria searchCriteria)
+    {
+        var pageableRequests = new IndexerPageableRequestChain();
+
+        pageableRequests.Add(GetPagedRequests($"{searchCriteria.SanitizedSearchTerm}", searchCriteria.Categories));
+
+        return pageableRequests;
+    }
+
+    private IEnumerable<IndexerRequest> GetPagedRequests(string term, int[] categories, string imdbId = null)
+    {
+        var parameters = new NameValueCollection
+        {
+            { "do", "search" },
+            { "category", "0" },
+            { "include_dead_torrents", "no" }
+        };
+
+        if (imdbId.IsNotNullOrWhiteSpace())
+        {
+            parameters.Set("search_type", "t_both");
+            parameters.Set("keywords", imdbId);
+        }
+        else
+        {
+            parameters.Set("search_type", "t_name");
+            parameters.Set("keywords", term);
+        }
+
+        var queryCats = _capabilities.Categories.MapTorznabCapsToTrackers(categories);
+
+        if (queryCats.Any())
+        {
+            parameters.Set("selectedcats2", string.Join(",", queryCats));
+        }
+
+        var searchUrl = _settings.BaseUrl + "browse.php";
+
+        var request = new IndexerRequest(searchUrl, HttpAccept.Html);
+        request.HttpRequest.SetContent(parameters.GetQueryString());
```

Comment:
XSpeeds expects this to be in form parameter format in the post request. Use RequestBuilder.AddFormParameter to add each parameter.